hwy/detect_targets.h

```diff
@@ -273,8 +263,7 @@
 #endif
 
 #if HWY_ARCH_PPC && (HWY_COMPILER_GCC || HWY_COMPILER_CLANG)
-#if defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && \
-    defined(__ALTIVEC__) && defined(__VSX__) && defined(__POWER8_VECTOR__) && \
+#if defined(__ALTIVEC__) && defined(__VSX__) && defined(__POWER8_VECTOR__) && \
```

Comment:
Eigen/xxhash use `__POWER8_VECTOR__,` seems to be what clang likes/reports. I am also seeing `__VSX__` in clang, libpng, abseil, libdav1d. Eigen checks for either `__ALTIVEC__` **or** `__VSX__.` So it seems those are both legit, but possibly clang-specific.

I'm open to suggestions as to what we should do on GCC. But perhaps it is safest to be conservative and only *assume* PPC8 if we see both ALTIVEC+VSX (because we use both); remember we can still use PPC8 via dynamic dispatch if it's actually supported.

Reply:
> Debian/ppc64el:
> 
> ```
> % c++   -Wp,-dM -E -c - < /dev/null | grep -Fi alti
> #define __ALTIVEC__ 1
> #define __pixel __attribute__((altivec(pixel__))) unsigned short
> #define __vector __attribute__((altivec(vector__)))
> #define __bool __attribute__((altivec(bool__))) unsigned
> #define __APPLE_ALTIVEC__ 1
> ```
> 
> and
> 
> ```
> % c++   -Wp,-dM -E -c - < /dev/null | grep -Fi _VSX__
> #define __VSX__ 1
> ```
> 
> but on Debian/ppc64:
> 
> ```
> % c++   -Wp,-dM -E -c - < /dev/null | grep -Fi alti
> #define __pixel __attribute__((altivec(pixel__))) unsigned short
> #define __vector __attribute__((altivec(vector__)))
> #define __bool __attribute__((altivec(bool__))) unsigned
> ```
> 
> I need to explicitly:
> 
> ```
> % c++ -maltivec   -Wp,-dM -E -c - < /dev/null | grep -Fi alti
> #define __ALTIVEC__ 1
> #define __pixel __attribute__((altivec(pixel__))) unsigned short
> #define __vector __attribute__((altivec(vector__)))
> #define __bool __attribute__((altivec(bool__))) unsigned
> #define __APPLE_ALTIVEC__ 1
> ```
> 
> `__VSX__` is never defined on ppc8/BE IMHO

The ```-mcpu=power8``` or ```-mpower8-vector``` option needs to be specified on the gcc/clang command line to enable the VSX instructions and POWER8 vector instructions on big-endian PPC targets.

Here are the various macros that are defined with different gcc options on big-endian PPC64:
```
$ powerpc64-linux-musl-gcc -dM -E -maltivec - </dev/null | egrep 'ALTIVEC|VSX|VECTOR'
#define __ALTIVEC__ 1
#define __APPLE_ALTIVEC__ 1
$ hwy_ppc9be_build$ powerpc64-linux-musl-gcc -dM -E -maltivec -mpower8-vector - </dev/null | egrep 'ALTIVEC|VSX|VECTOR'
#define __VSX__ 1
#define __ALTIVEC__ 1
#define __APPLE_ALTIVEC__ 1
#define __POWER8_VECTOR__ 1
$ powerpc64-linux-musl-gcc -dM -E -mcpu=power8 - </dev/null | egrep 'ALTIVEC|VSX|VECTOR'
#define __VSX__ 1
#define __ALTIVEC__ 1
#define __APPLE_ALTIVEC__ 1
#define __POWER8_VECTOR__ 1
```

The ```defined(__VSX__)``` and ```defined(__POWER8_VECTOR__)``` checks are actually needed as some of the VSX and POWER8 vector intrinsics are not enabled on GCC/Clang unless both the ```__VSX__``` and ```__POWER8_VECTOR___``` macros are defined.

The altivec.h header on both GCC and clang also actually checks to see if the ```__VSX__``` macro is defined, and the altivec.h header also checks to see if the ```__POWER8_VECTOR__``` macro is defined.

There will also be compilation errors without the checks for the  ```defined(__VSX__)``` and ```defined(__POWER8_VECTOR__)``` macros on GCC/Clang if __VSX__ and __POWER8_VECTOR__ are not defined unless the HWY_PPC8/HWY_PPC9/HWY_PPC10 targets are disabled through HWY_DISABLED_TARGETS, HWY_COMPILE_ONLY_SCALAR, or HWY_COMPILE_ONLY_EMU128.

The ```-mcpu=power8``` and ```-mpower8-vector``` options are available on GCC 4.9 and later.

The reason why the HWY_PPC8/HWY_PPC9/HWY_PPC10 targets are not enabled by default on big-endian PPC64 is that the VSX and POWER8 vector instructions need to be enabled through the ```-mcpu=power8``` and ```-mpower8-vector``` options.

Note that programs compiled with the ```-mcpu=power8``` or ```-mpower8-vector``` options may fail to run on PPC CPU's that don't support the POWER8 instruction set.

Reply:
Sounds good, thanks @johnplatts and @malaterre .